CMakeLists.txt

```diff
@@ -463,7 +465,7 @@ if (MINGW)
   target_link_libraries(mfem ws2_32)
 endif()
 set_target_properties(mfem PROPERTIES VERSION "${mfem_VERSION}")
-set_target_properties(mfem PROPERTIES SOVERSION "${mfem_VERSION}")
+set_target_properties(mfem PROPERTIES SOVERSION 4)
```

Comment:
Why can't this be `${mfem_VERSION}`? The hardcoded version number is not good.

Reply:
Because it used the number 4.2.1 and since it was identical with the actual .so file it wouldn't produce symlinks. I know we can use something like `VERSION_MAJOR` but I will have to familiarize my self with the exact syntax.

Reply:
[Here](https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION_MAJOR.html) I found this variable
```
PROJECT_VERSION_MAJOR
```

Reply:
Using `PROJECT_VERSION_MAJOR` doesn't work.
```
/usr/lib64/libmfem.so -> libmfem.so.
/usr/lib64/libmfem.so. -> libmfem.so.4.2.1
/usr/lib64/libmfem.so.4.2.1

```
No "4" after the dot.

Reply:
Probably because the mfem version is defined wrong in the cmake buildsystem. Could you verify that the SEMVER is cmake conform?

Reply:
> Probably because the mfem version is defined wrong in the cmake buildsystem. Could you verify that the SEMVER is cmake conform?

I was about to write this. Do you want me to do this like this?
```
project(mfem 4.2.1)
```

Using 
```
project(mfem NONE)
# Current version of MFEM, see also `makefile`.
#   mfem_VERSION = (string)
#   MFEM_VERSION = (int)   [automatically derived from mfem_VERSION]
set(${PROJECT_NAME}_VERSION 4.2.1)
``` 
doesn't seem the recommended way to go.

Reply:
What is SEMVER?

Reply:
Yes, this is a custom version decomposition. We should figure out how to use `project(mfem 4.2.1)` and have an `mfem_VERSION` (string) and `MFEM_VERSION` (int) derived from the cmake generated variable.

Reply:
We can use both. The way you devs are using it seems ok for you and we will use the recommended too. At every version bump the version will change in two places.